File: src/main/java/io/vproxy/vfx/ui/button/TransparentFusionImageButton.java
package io.vproxy.vfx.ui.button;

import io.vproxy.vfx.theme.Theme;
import javafx.scene.image.Image;
import javafx.scene.paint.Color;

public class TransparentFusionImageButton extends FusionImageButton {
    public TransparentFusionImageButton() {
    }

    public TransparentFusionImageButton(Image image) {
        super(image);
    }

    @Override
    protected Color normalColor() {
        return Theme.current().transparentFusionButtonNormalBackgroundColor();
    }

    @Override
    protected Color hoverColor() {
        return Theme.current().transparentFusionButtonHoverBackgroundColor();
    }

    @Override
    protected Color downColor() {
        return Theme.current().transparentFusionButtonDownBackgroundColor();
    }
}

File: src-ts/src/ui/button/TransparentFusionImageButton.ts
import { Color } from '../../javafx/color.js';
import { FXImage } from '../../javafx/ImageView.js';
import { Theme } from '../../theme/Theme.js';
import { FusionImageButton } from './FusionImageButton.js';

export class TransparentFusionImageButton extends FusionImageButton {
  constructor();
  constructor(image: FXImage | null);
  constructor(image: FXImage | null = null) {
    super(image);
  }

  protected override normalColor(): Color {
    return Theme.current().transparentFusionButtonNormalBackgroundColor();
  }

  protected override hoverColor(): Color {
    return Theme.current().transparentFusionButtonHoverBackgroundColor();
  }

  protected override downColor(): Color {
    return Theme.current().transparentFusionButtonDownBackgroundColor();
  }
}
